pkg/controllers/machine_controller.go

```diff
@@ -173,6 +189,118 @@ func (r *MachineReconciler) Start(ctx context.Context) error {
 	return nil
 }
 
+func (r *MachineReconciler) startGarbageCollector(ctx context.Context, log logr.Logger) {
+	wait.UntilWithContext(ctx, func(ctx context.Context) {
+		machines, err := r.machines.List(ctx)
+		if err != nil {
+			log.Error(err, "failed to list machines")
+			return
+		}
+
+		for _, machine := range machines {
+			if !slices.Contains(machine.Finalizers, MachineFinalizer) || machine.DeletedAt == nil {
+				continue
+			}
+
+			logger := log.WithValues("machineId", machine.ID)
+			var destroid bool
+
+			if !machine.FirstShutdownAt.IsZero() && time.Now().After(machine.FirstShutdownAt.Add(r.vmGracefulShutdownTimeout)) {
+				destroid, err = r.destroyAndCleanupMachine(ctx, logger, machine)
+			} else {
+				err = r.processShutdown(ctx, logger, machine)
+			}
+
+			if destroid || libvirt.IsNotFound(err) {
+				err = r.cleanupAdditionalResources(ctx, logger, machine)
+				if err == nil {
+					continue
+				}
+			}
+
+			if err != nil {
+				logger.Error(err, "failed to garbage collector process")
+			}
+		}
+
+	}, r.resyncIntervalGarbageCollector)
+}
+
+func (r *MachineReconciler) destroyAndCleanupMachine(ctx context.Context, log logr.Logger, machine *api.Machine) (bool, error) {
```

Comment:
rename function. cleanup isn't anymore part of this function

Reply:
done.